cluster/cluster.go

```diff
@@ -40,6 +45,77 @@ func Join(ctx context.Context, cfg Config) (*Cluster, error) {
 	}, nil
 }
 
+type MemberAddInfo struct {
+	InitialCluster      string
+	InitialClusterState string
+}
+
+func (c *Cluster) MemberAdd(ctx context.Context, name, peerURL string) (*MemberAddInfo, error) {
+	etcdCfg := c.etcd.Config()
+	cfg := clientv3.Config{
+		Endpoints:   []string{etcdCfg.LCUrls[0].String()},
+		DialTimeout: 5 * time.Second,
+	}
+	client, err := clientv3.New(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create etcd client from config: %w", err)
+	}
+
+	mresp, err := client.MemberAdd(ctx, []string{peerURL})
+	if err != nil {
+		return nil, fmt.Errorf("failed to add member with peerURL n%v: %w", peerURL, err)
+	}
+
+	initialClusterStrings := make([]string, 2)
+	for _, member := range mresp.Members {
+		if member.Name == "" && strings.Compare(peerURL, member.PeerURLs[0]) == 0 {
+			initialClusterStrings = append(initialClusterStrings, initialClusterStringFormatter(name, member.PeerURLs[0]))
+		} else {
+			initialClusterStrings = append(initialClusterStrings, initialClusterStringFormatter(member.Name, member.PeerURLs[0]))
+		}
+	}
```

Comment:
Let me know if I understand this correctly. 

So we are making a request to a node in the initial cluster. It will return ALL members that are currently in the cluster. All these members are required to be in the inital cluster field for the new node.

Reply:
Yeah, it returns all members + new member

Reply:
I can do some additional testing to see if you only need the original initial cluster, but I don't see anywhere in the etcd docs that indicate that that's the case, + it wouldn't make sense if say member1 adds itself by adding itself to the initial cluster and then member2 adds itself but ignores the fact that member1 added itself to the initial cluster configuration

Reply:
> + it wouldn't make sense if say member1 adds itself by adding itself to the initial cluster and then member2 adds itself but ignores the fact that member1 added itself to the initial cluster configuration

I would've imagined it to work with something along these lines. Like each new node simply asks the nodes in the initial cluster to join, and that information would be propagated through the initial cluster to the other nodes. But we can leave this testing as another issue.